src/compiler/util.h

```diff
@@ -122,6 +122,27 @@ class PathBuilder {
     join(segment4);
   }
 
+  // Joins the given sub_path to this builder.
+  // The sub_path must use slashes as path separators.
+  void join_slash_path(const std::string& sub_path) {
+    if (sub_path.empty()) return;
+    size_t start_pos = 0;
+    while (start_pos < sub_path.size()) {
+      auto slash_pos = sub_path.find_first_of('/', start_pos);
+      if (slash_pos == std::string::npos) {
+        // No more slashes, use the rest of the string.
+        slash_pos = sub_path.size();
+      }
+      // For absolute paths we need to add the leading "/".
+      if (slash_pos == 0 && buffer_.empty()) {
+        buffer_ += fs_->path_separator();
+      } else if (slash_pos > start_pos) {
```

Comment:
Maybe explain the guard against `slash_pos == start_pos` in a comment? Are you collapsing repeated slashes?

Reply:
I removed that 'if'.
I don't think we should ever get multiple '/' and there is no reason we should pick one over the other. Also, we do a canonicalize later, which would remove it anyway.